// @dicebear/avatars-human-sprites/dist/index.es.js
import { utils } from '@dicebear/avatars';
import { create as create$1 } from '@dicebear/avatars-male-sprites';
import { create as create$2 } from '@dicebear/avatars-female-sprites';

var schema$1 = {
  "title": "Options",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": {
    "mood": {
      "title": "Mood",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["happy", "sad", "surprised"]
      }
    }
  },
  "additionalProperties": false
};

var style = {
  meta: {
    title: 'Human / 8biticon',
    creator: 'Plastic Jam',
    contributor: 'Florian Körner',
    source: 'https://github.com/dicebear/dicebear',
    license: {
      name: 'MIT',
      url: 'https://github.com/dicebear/dicebear/blob/v4/packages/avatars-human-sprites/LICENSE'
    }
  },
  schema: schema$1,
  create: function create(_ref) {
    var prng = _ref.prng,
        options = _ref.options;

    if (prng.bool(50)) {
      return create$1({
        prng: prng,
        options: options
      });
    } else {
      return create$2({
        prng: prng,
        options: options
      });
    }
  }
};

/*!
 * DiceBear Human (@dicebear/avatars-human-sprites)
 *
 * Code licensed under MIT (https://github.com/dicebear/dicebear/blob/v4/packages/avatars-human-sprites/LICENSE)
 * Copyright (c) 2012 Plastic Jam
 * Copyright (c) 2021 Florian Körner
 */
var create = style.create,
    meta = style.meta,
    schema = style.schema;
/** @deprecated will be removed in Version 5.0 */

var index = utils.style.createLegacyWrapper({
  create: create,
  meta: meta,
  schema: schema
});

export { create, index as default, meta, schema };
